endpoints/put.go

```diff
@@ -90,7 +90,21 @@ func NewPutHandler(backend backends.Backend, maxNumValues int) func(http.Respons
 			resps.Responses[i].UUID = uuid.NewV4().String()
 			ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
 			defer cancel()
-			err = backend.Put(ctx, resps.Responses[i].UUID, toCache, p.TTLSeconds)
+			// Only allow setting a provided key if configured (and ensure a key is provided).
+			if allowKeys && len(p.Key) > 0 {
+				s, err := backend.Get(ctx, p.Key)
+				if err != nil || len(s) == 0 {
+					resps.Responses[i].UUID = p.Key
+				} else {
+					resps.Responses[i].UUID = ""
+				}
+			}
+			// If we have a blank UUID, don't store anything.
+			// Eventually we may want to provide error details, but as of today this is the only non-fatal error
+			// Future error details could go into a second property of the Responses object, such as "errors"
+			if len(resps.Responses[i].UUID) > 0 {
+				err = backend.Put(ctx, resps.Responses[i].UUID, toCache, p.TTLSeconds)
```

Comment:
The code that checks the value of err beneath the following curly brace should be moved up above the curly brace, as the check should be performed only if the Put() is performed.

Reply:
Good catch, thanks,